src/userlist/userListReducers.js

```diff
@@ -31,46 +26,57 @@ type PresenceMap = {
 //   presence: PresenceMap,
 // }
 
-const initialState = fromJS({
-  filter: '',
-  users: [],
-});
+const priorityToState = {
+  0: 'offline',
+  1: 'idle',
+  2: 'active',
+};
+
+const stateToPriority = {
+  offline: 0,
+  idle: 1,
+  active: 2,
+};
+
+export const activityFromPresence = (presence: Presence): UserStatus =>
+  priorityToState[Math.max(...Object.values(presence).map(x => stateToPriority[x.status]))];
 
-const activityFromPresence = (presence: PresenceMap): UserStatus =>
-  'active';
+export const timestampFromPresence = (presence: Presence): UserStatus =>
+  Math.max(...Object.values(presence).map(x => x.timestamp));
 
-const reducer = (state = initialState, action) => {
+export const activityFromTimestamp = (activity: string, timestamp: number) =>
+  ((new Date() / 1000) - timestamp > 60 ? 'offline' : activity);
+
+const initialState = fromJS([]);
+
+export default (state = initialState, action) => {
   switch (action.type) {
     case PRESENCE_RESPONSE: {
-      let newState = state;
-      Object.keys(action.presence).forEach(x => {
-        const status = activityFromPresence(action.presence[x]);
-        const user = state.find(u => u.email === x);
-        newState = state.setIn([user, { status }]);
-      });
-      return newState;
+      return Object.keys(action.presence).reduce((currentState, email) => {
+        const userIndex = state.findIndex(u => u.get('email') === email);
+        if (userIndex === -1) return currentState;
+
+        const p = action.presence[email];
+        const timestamp = timestampFromPresence(p);
+        const status = activityFromTimestamp(activityFromPresence(p), timestamp);
+        return currentState.setIn([userIndex, 'status'], status).setIn([userIndex, 'timestamp'], timestamp);
+      }, state);
+    }
+    case GET_USER_RESPONSE: {
+      const users = action.users.map(x => ({
```

Comment:
Can we rename `x` to `user`? I think it's clearer to avoid short variable names like `x` unless it's a loop index or a really short function